codec/src/config.rs

```diff
@@ -2,37 +2,97 @@
 
 use core::ops::{Bound, RangeBounds};
 
-/// Configuration for limiting the range of a [usize] value.
+/// Configuration for limiting the range of a value.
 ///
 /// This is often used to configure length limits for variable-length types or collections.
 ///
-/// # Example
+/// # Examples
 ///
 /// ```
 /// use commonware_codec::RangeCfg;
 ///
-/// // Limit lengths to 0..=1024
-/// let cfg = RangeCfg::from(0..=1024);
+/// // Limit lengths to 0..=1024 (type inferred as usize)
+/// let cfg = RangeCfg::new(0..=1024);
 /// assert!(cfg.contains(&500));
 /// assert!(!cfg.contains(&2000));
 ///
 /// // Allow any length >= 1
 /// let cfg_min = RangeCfg::from(1..);
 /// assert!(cfg_min.contains(&1));
 /// assert!(!cfg_min.contains(&0));
+///
+/// // Works with other integer types
+/// let cfg_u8: RangeCfg<u8> = RangeCfg::new(0u8..=255u8);
+/// assert!(cfg_u8.contains(&128));
+///
+/// let cfg_u32 = RangeCfg::new(0u32..1024u32);
+/// assert!(cfg_u32.contains(&500));
 /// ```
 #[derive(Clone, Copy, Debug, Eq, PartialEq, Hash)]
-pub struct RangeCfg {
+pub struct RangeCfg<T: Copy + PartialOrd> {
     /// The lower bound of the range.
-    start: Bound<usize>,
+    start: Bound<T>,
 
     /// The upper bound of the range.
-    end: Bound<usize>,
+    end: Bound<T>,
+}
+
+impl<T: Copy + PartialOrd> From<core::ops::Range<T>> for RangeCfg<T> {
```

Comment:
This PR removes the blanket `impl<R: RangeBounds<usize>> From<R> for RangeCfg`.

Instead of having a blanket implementation, we have implementations for the mostly commonly used concrete types.